src/webui/src/components/Modal/CustomizedTrial.tsx

```diff
@@ -0,0 +1,329 @@
+                        val === customized[item]) === undefined) {
+                        flag = true;
+                        return;
+                    }
+                } else {
+                    if (customized[item] < searchSpace[item]._value[0]
+                        || customized[item] > searchSpace[item]._value[1]) {
+                        flag = true;
+                        return;
+                    }
+                }
+            }
+        });
+        if (flag !== false) {
+            // open the warning modal
+            if (this._isCustomizeMount === true) {
+                this.setState(() => ({ isShowWarning: true, customParameters: customized }));
+            }
+        } else {
+            // submit a customized job
+            this.submitCustomize(customized);
+        }
+
+    }
+
+    warningConfirm = () => {
+        if (this._isCustomizeMount === true) {
+            this.setState(() => ({ isShowWarning: false }));
+        }
+        const { customParameters } = this.state;
+        this.submitCustomize(customParameters);
+    }
+
+    warningCancel = () => {
+        if (this._isCustomizeMount === true) {
+            this.setState(() => ({ isShowWarning: false }));
+        }
+    }
+
+    submitCustomize = (customized: Object) => {
+        // delete `tag` key
+        for (let i in customized) {
+            if (i === 'tag') {
+                delete customized[i];
+            }
+        }
+        axios(`${MANAGER_IP}/trial-jobs`, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            data: customized
+        })
+            .then(res => {
+                if (res.status === 200) {
+                    if (this._isCustomizeMount === true) {
+                        this.setState(() => ({ customID: res.data.sequenceId , isShowSubmitSucceed: true }));
+                        this.props.closeCustomizeModal();
+                    }
+                } else {
+                    if (this._isCustomizeMount === true) {
+                        this.setState(() => ({ isShowSubmitFailed: true }));
+                    }
+                }
+            })
+            .catch(error => {
+                if (this._isCustomizeMount === true) {
+                    this.setState(() => ({ isShowSubmitFailed: true }));
+                }
+            });
+    }
+
+    closeSucceedHint = () => {
+        // also close customized trial modal
+        if (this._isCustomizeMount === true) {
+            this.setState(() => ({ isShowSubmitSucceed: false }));
+            this.props.closeCustomizeModal();
+        }
+    }
+
+    closeFailedHint = () => {
+        // also close customized trial modal
+        if (this._isCustomizeMount === true) {
+            this.setState(() => ({ isShowSubmitFailed: false }));
+            this.props.closeCustomizeModal();
+        }
+    }
+
+    componentDidMount() {
+        this._isCustomizeMount = true;
```

Comment:
Official react doc says we should not check "is mounted" to update states.
Instead we should cancel handlers on unmount.
If there is no handler, than this check does nothing.

Reply:
OK.